### PR TITLE
Various fixes gogc

### PR DESCRIPTION
**Pull request by vpiyush**
Improvements On
1. GOGC setting
2. proposal verification 
3. keep a counter to avoid iterating the pending map every time we reorg

### CODE REVIEW

File: core/tx_pool.go [hunk 1]
```diff
@@ -1385,10 +1388,7 @@ func (pool *TxPool) promoteExecutables(accounts []common.Address) []*types.Trans
 // pending limit. The algorithm tries to reduce transaction counts by an approximately
 // equal number for all for accounts with many pending transactions.
 func (pool *TxPool) truncatePending() {
-	pending := uint64(0)
-	for _, list := range pool.pending {
-		pending += uint64(list.Len())
-	}
+	pending := uint64(pool.totalPending.Load())
```
Comment (lorenzo-dev1) [inline core/tx_pool.go:1391]: I would be more confident in approving this change if we can have a test that ensures that `pool.totalPending` is always equal to the length computed via the legacy iteration method

Comment (lorenzo-dev1) [reply core/tx_pool.go:1391]: Also the `pool.stats()` function still uses the legacy method

Comment (lorenzo-dev1) [reply core/tx_pool.go:1391]: Another idea would be to keep both methods for now, and compare the values when the len is computed. If they mismatch print a log. That way we can monitor if the two values ever mismatch and eventually update for next release.

**Review by Jason-Zhangxin-Chen — APPROVED** [review]
LGTM.

File: core/tx_pool.go (hunk 1 above)
Comment (vpiyush) [reply core/tx_pool.go:1388]: updated pool.stats to use the totalPending counter, All of the tests use to validate the pool internals . this should automatically test the totalPending counter

**Comment by lorenzo-dev1** [comment]
The txpool change looks definitely better now that `Stats()` is modified and therefore we are testing the change. However tbh I am not very familiar with the txpool logic, therefore I don't really feel super confident about it. I would just rely on hoping that the go-ethereum tests (which I never looked at) are properly written to approve...

What is your opinion @yazzaoui ?

**Review by yazzaoui — APPROVED** [review]

**Pull request merged**